src/System.Data.Common/src/System/Data/Common/DbDataReader.cs

```diff
@@ -0,0 +1,297 @@
+
+        abstract public int GetOrdinal(string name);
+
+
+        abstract public bool GetBoolean(int ordinal);
+
+        abstract public byte GetByte(int ordinal);
+
+        abstract public long GetBytes(int ordinal, long dataOffset, byte[] buffer, int bufferOffset, int length);
+
+        abstract public char GetChar(int ordinal);
+
+        abstract public long GetChars(int ordinal, long dataOffset, char[] buffer, int bufferOffset, int length);
+
+        public DbDataReader GetData(int ordinal)
+        {
+            return GetDbDataReader(ordinal);
+        }
+
+
+        virtual protected DbDataReader GetDbDataReader(int ordinal)
+        {
+            // NOTE: This method is virtual because we're required to implement
+            //       it however most providers won't support it. Only the OLE DB 
+            //       provider supports it right now, and they can override it.
+            throw ADP.NotSupported();
+        }
+
+        abstract public DateTime GetDateTime(int ordinal);
+
+        abstract public Decimal GetDecimal(int ordinal);
+
+        abstract public double GetDouble(int ordinal);
+
+        abstract public float GetFloat(int ordinal);
+
+        abstract public Guid GetGuid(int ordinal);
+
+        abstract public Int16 GetInt16(int ordinal);
+
+        abstract public Int32 GetInt32(int ordinal);
+
+        abstract public Int64 GetInt64(int ordinal);
+
+        virtual public Type GetProviderSpecificFieldType(int ordinal)
+        {
+            // NOTE: This is virtual because not all providers may choose to support
+            //       this method, since it was added in Whidbey.
+            return GetFieldType(ordinal);
+        }
+
+        virtual public Object GetProviderSpecificValue(int ordinal)
+        {
+            // NOTE: This is virtual because not all providers may choose to support
+            //       this method, since it was added in Whidbey
+            return GetValue(ordinal);
+        }
+
+        virtual public int GetProviderSpecificValues(object[] values)
+        {
+            // NOTE: This is virtual because not all providers may choose to support
+            //       this method, since it was added in Whidbey
+            return GetValues(values);
+        }
+
+        abstract public String GetString(int ordinal);
+
+        virtual public Stream GetStream(int ordinal)
+        {
+            using (MemoryStream bufferStream = new MemoryStream())
+            {
+                long bytesRead = 0;
+                long bytesReadTotal = 0;
+                byte[] buffer = new byte[4096];
+                do
+                {
+                    bytesRead = GetBytes(ordinal, bytesReadTotal, buffer, 0, buffer.Length);
+                    bufferStream.Write(buffer, 0, (int)bytesRead);
+                    bytesReadTotal += bytesRead;
+                } while (bytesRead > 0);
+
+                return new MemoryStream(bufferStream.ToArray(), false);
```

Comment:
Why copy to another memory stream rather than just returning bufferStream, i.e.
```C#
public virtual Stream GetStream(int ordinal)
{
    var bufferStream = new MemoryStream();
    ... // copy loop
    bufferStream.Position = 0;
    return bufferStream;
}
```